csharp/src/Apache.Arrow/Arrays/ArrayData.cs

```diff
@@ -22,9 +22,9 @@ namespace Apache.Arrow
     public sealed class ArrayData
     {
         public readonly IArrowType DataType;
-        public readonly int Length;
```

Comment:
I think these can remain `readonly` can't they?

I'd be surprised if someone changed the Length of an array instance.

Reply:
Array.Slice is manipulating it. The way I look at it, ArrayData is just a view over the same underlying buffers, so if someone wants a slice, "Length" needs to change. We could still keep it readonly and provide a set method(which is what I started out with originally), but I thought this was more succinct.